File: MoodByte/src/main/java/org/example/ModeloDTO/UsuarioDTO.java
package org.example.ModeloDTO;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import org.example.Modelo.Genero;
import org.example.Modelo.TipoUsuario;
import org.example.Modelo.Usuario;

import java.time.LocalDate;

public record UsuarioDTO(
        long id,
        String nombreCompleto,
        String nombreUsuario,
        String password,
        int edad,
        @Enumerated(EnumType.STRING)
        Genero genero,
        @Enumerated(EnumType.STRING)
        TipoUsuario tipoUsuario,
        LocalDate fechaRegistro,
        LocalDate fechaNacimiento,
        int nivel,

        double expAcumulada) {
    public UsuarioDTO(Usuario usuario){
        this(
          usuario.getId(),
          usuario.getNombreCompleto(),
          usuario.getNombreUsuario(),
                usuario.getPassword(),
                usuario.getEdad(),
                usuario.getGenero(),
                usuario.getTipoUsuario(),
                usuario.getFechaRegistro(),
                usuario.getFechaNacimiento(),
                usuario.getNivel(),
                usuario.getExpAcumulada()
        );
    }
}

File: MoodByte_App/app/src/main/java/com/example/moodbyte/data/remote/dtos/UsuarioDto.kt
package com.example.moodbyte.data.remote.dtos

import com.example.moodbyte.data.local.entities.UsuarioEntity

data class UsuarioDto(
    val id: Long?,
    val nombreCompleto: String,
    val nombreUsuario: String,
    val password: String,
    val edad: Int,
    val genero: String,
    val tipoUsuario: String,
    val fechaRegistro: String,
    val fechaNacimiento: String,
    val nivel: Int,
    val expAcumulada: Double
)

fun UsuarioDto.toEntity() = UsuarioEntity(
    id = id,
    nombreCompleto = nombreCompleto,
    nombreUsuario = nombreUsuario,
    password = password,
    edad = edad,
    genero = genero,
    tipoUsuario = tipoUsuario,
    fechaRegistro = fechaRegistro,
    fechaNacimiento = fechaNacimiento,
    nivel = nivel,
    expAcumulada = expAcumulada
)